cudf/backend.py

```diff
@@ -0,0 +1,135 @@
+"""Support different CUDA backends: numba.cuda, pyarrow.cuda.
+
+Usage:
+  from cudf.backend import cuda
+  cuda.use_numba()    # to enable numba.cuda
+  cuda.use_arrow()    # to enable pyarrow.cuda
+"""
+# Author: Pearu Peterson
+# Created: October 2018
+
+import numpy as np
+from .cudaarray import CudaNDArray
+
+
+class NumbaCudaDeviceArray:
+
+    def __init__(self, parent):
+        self.parent = parent
+        self.cuda = parent.cuda
+
+    def is_cuda_ndarray(self, obj):
+        return hasattr(obj, '__cuda_array_interface__')
+
+    def DeviceNDArray(self, *args, **kwargs):
```

Comment:
Should this be `device_ndarray()`?

Reply:
No, it is a part of backend switching magic.

`DeviceNDArray` is a class implemented within numba.cuda. When pyarrow.cuda backend is effective, any code that would call `numba.cuda.devicearray.DeviceNDArray`, will call the above `DeviceNDArray` method that returns equivalent pyarrow.cuda object.